### ISSUE TITLE
Insufficient space when lv size matches vg

### ISSUE THREAD

**Issue by tabowling** [linux-system-roles/storage#13]
Playbook fails when lv size is equal to vg size.

```
TASK [linux-system-roles.storage : Make sure LV exists] ***************************************************************************************************************************
fatal: [rhel7]: FAILED! => {"changed": false, "err": "  Volume group \"composer\" has insufficient free space (5119 extents): 5120 required.\n", "msg": "Creating logical volume 'composer' failed", "rc": 5}

[root@rhel7 ~]# vgs
  VG       #PV #LV #SN Attr   VSize   VFree  
  composer   1   0   0 wz--n- <20.00g <20.00g
  rhel       1   2   0 wz--n-  <7.00g      0 
```

Playbook used was:

```
- hosts: all
  become: yes
  become_method: sudo
  become_user: root

  vars:
    use_partitions: false
  tasks:
    - name: Configure Composer Storage
      include_role:
        name: linux-system-roles.storage
      vars:
        storage_pools:
          - name: composer
            disks: ['vdb']
            # type: lvm
            state: present
            volumes:
              - name: composer
                size: 20G
                # type: lvm
                # fs_type: xfs
                fs_label: "composer"
                mount_point: '/var/lib/lorax/composer'
```

**Comment by dwlehman** [comment]
IMO this is an LVM bug, and they've agreed there should be an option to interpret size as "up to this size" for several years now, but no code has been written. We can try to handle this in the role, but it will take a bit of work (which is fraught with danger due to guesswork).

**Comment by pcahyna** [comment]
Is there any bug actually? The output `<20.00g` indicates there is not exactly 20g of space, it is rounded and there is actually less space than that. I think it is better to let the role fail than let it create a smaller volume than what you asked for.

**Comment by dwlehman** [comment]
The percentage-based size work (#39) will address this issue in another way, but I am also open to truncating sizes as needed up to some agreed-upon percentage to make things work seamlessly without requiring the user to predict the total space used by metadata .

When I wrote blivet many years ago I was determined to create volumes/devices the exact size requested by the user. Over the years I have gradually shifted to the view that it does not strictly matter for any tangible reason whether you get the full size you request or whether the volume you get is actually short -- even by as much as 100 MiB in most cases. Of course not everyone will be pleased by such a convention, but in this role we have a mandate to make it easy to use for the majority of cases. We could also introduce a switch to enable/disable size fuzzing if we want to try to please everyone.

**Comment by vojtechtrefny** [comment]
Closed via #199
